src/config.rs

```diff
@@ -0,0 +1,1191 @@
+    pub pidfile: Option<PathBuf>,
+
+    pub policies: Vec<Policy>,
+
+    /// Adds a directory to search for .service files,
+    /// which tell the dbus-daemon how to start a program to provide a particular well-known bus
+    /// name.
+    #[serde(default)]
+    pub servicedirs: Vec<PathBuf>,
+
+    /// Specifies the setuid helper that is used to launch system daemons with an alternate user.
+    pub servicehelper: Option<PathBuf>,
+
+    /// If `true`, the bus daemon will log to syslog.
+    pub syslog: bool,
+
+    /// This element only controls which message bus specific environment variables are set in
+    /// activated clients.
+    pub r#type: Option<BusType>,
+
+    /// The user account the daemon should run as, as either a username or a UID.
+    /// If the daemon cannot change to this UID on startup, it will exit.
+    /// If this element is not present, the daemon will not change or care about its UID.
+    pub user: Option<String>,
+}
+
+impl TryFrom<Document> for Config {
+    type Error = Error;
+
+    fn try_from(value: Document) -> std::result::Result<Self, Self::Error> {
+        let mut bc = Config::default();
+
+        for element in value.busconfig {
+            match element {
+                Element::AllowAnonymous => bc.allow_anonymous = true,
+                Element::Auth(s) => {
+                    bc.auth.insert(s);
+                }
+                Element::Fork => bc.fork = true,
+                Element::Include(_) => {
+                    // NO-OP: removed during `Document::resolve_includes`
+                }
+                Element::Includedir(_) => {
+                    // NO-OP: removed during `Document::resolve_includedirs`
+                }
+                Element::KeepUmask => bc.keep_umask = true,
+                Element::Limit => {
+                    // NO-OP: deprecated and ignored
+                }
+                Element::Listen(s) => {
+                    bc.listen.insert(s);
+                }
+                Element::Pidfile(p) => bc.pidfile = Some(p),
+                Element::Policy(pe) => {
+                    if let Some(p) = OptionalPolicy::try_from(pe)? {
+                        bc.policies.push(p);
+                    }
+                }
+                Element::Servicedir(p) => {
+                    bc.servicedirs.push(p);
+                }
+                Element::Servicehelper(p) => {
+                    // NOTE: we're assuming this has the same "last one wins" behaviour as `<type>`
+                    // TODO: warn and then ignore if we aren't reading:
+                    // /usr/share/dbus-1/system.conf
+                    bc.servicehelper = Some(p);
+                }
+                Element::StandardSessionServicedirs => {
+                    // NOTE: because servicedir sequence is important, we cannot de-duplicate this
+                    // TODO: warn and then ignore if we aren't reading: /etc/dbus-1/session.conf
```

Comment:
They aren't related

Although I think not having any deduplication and also not mentioning it at all probably makes more sense that that note

Reply:
Thanks. It's still a bit hard to understand what de-duplication is this about. Also it's not clear why we should warn and ignore.

Reply:
https://dbus.freedesktop.org/doc/dbus-daemon.1.html#configuration_file

> The `<standard_session_servicedirs/>` option is only relevant to the per-user-session bus daemon defined in /etc/dbus-1/session.conf. Putting it in any other configuration file would probably be nonsense.

So, my TODO here is really going above and beyond to protect the user from a footgun

But, it's not strictly in the specification

Reply:
And when I first went through the specification, it seemed silly to me that a configuration might contain this directive multiple times

But because search sequence matters, it didn't seem obvious how to deduplicate in a way that preserved user intent

What even would be their intent? 🤷

Reply:
Ah ok, thanks for explaining. I really don't think we need to care about duplication here. I'd just remove the comment about de-duplication.